homeassistant/components/mill/climate.py

```diff
@@ -52,11 +49,11 @@
 )
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the Mill heater."""
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Mill climate."""
     mill_data_connection = Mill(
-        config[CONF_USERNAME],
-        config[CONF_PASSWORD],
+        entry.data[CONF_USERNAME],
+        entry.data[CONF_PASSWORD],
         websession=async_get_clientsession(hass),
     )
     if not await mill_data_connection.connect():
```

Comment:
We should raise `ConfigEntryNotReady` here so it will retry in a bit once the connection might be established.